pkg/db/db_test.go

```diff
@@ -1592,6 +1592,137 @@ func TestReadWriteEslEvent(t *testing.T) {
+					EventJson: string(`{"env":"dev","app":"my-app","lockId":"ui-v2-ke1up","message":"test","metadata":{"authorEmail":"testemail@example.com","authorName":"testauthor"}}`),
+					Created:   time.Now(),
+					EslId:     1,
+				},
+				{
+					EventType: EvtCreateEnvironmentApplicationLock,
+					EventJson: string(`{"env":"dev2","app":"my-app","lockId":"ui-v2-ke1up","message":"test","metadata":{"authorEmail":"testemail@example.com","authorName":"testauthor"}}`),
+					Created:   time.Now(),
+					EslId:     2,
+				},
+				{
+					EventType: EvtCreateEnvironment,
+					EventJson: string(`{"env":"dev3","app":"my-app","lockId":"ui-v2-ke1up","message":"test","metadata":{"authorEmail":"testemail@example.com","authorName":"testauthor"}}`),
+					Created:   time.Now(),
+					EslId:     3,
+				},
+			},
+			Limit: 3,
+		},
+		{
+			Name: "More than limit",
+			Events: []EslEventRow{
+				{
+					EventType: EvtCreateApplicationVersion,
+					EventJson: string(`{"env":"dev","app":"my-app","lockId":"ui-v2-ke1up","message":"test","metadata":{"authorEmail":"testemail@example.com","authorName":"testauthor"}}`),
+					Created:   time.Now(),
+					EslId:     1,
+				},
+				{
+					EventType: EvtCreateEnvironmentGroupLock,
+					EventJson: string(`{"env":"dev2","app":"my-app","lockId":"ui-v2-ke1up","message":"test","metadata":{"authorEmail":"testemail@example.com","authorName":"testauthor"}}`),
+					Created:   time.Now(),
+					EslId:     2,
+				},
+				{
+					EventType: EvtCreateEnvironment,
+					EventJson: string(`{"env":"dev3","app":"my-app","lockId":"ui-v2-ke1up","message":"test","metadata":{"authorEmail":"testemail@example.com","authorName":"testauthor"}}`),
+					Created:   time.Now(),
+					EslId:     3,
+				},
+			},
+			Limit: 2,
+		},
+		{
+			Name: "Less than limit",
+			Events: []EslEventRow{
+				{
+					EventType: EvtCreateApplicationVersion,
+					EventJson: string(`{"env":"dev","app":"my-app","lockId":"ui-v2-ke1up","message":"test","metadata":{"authorEmail":"testemail@example.com","authorName":"testauthor"}}`),
+					Created:   time.Now(),
+					EslId:     1,
+				},
+			},
+			Limit: 3,
+		},
+	}
+
+	for _, tc := range tcs {
+		tc := tc
+		t.Run(tc.Name, func(t *testing.T) {
+			t.Parallel()
+			ctx := testutil.MakeTestContext()
+			dbHandler := setupDB(t)
+			err := dbHandler.WithTransaction(ctx, true, func(ctx context.Context, transaction *sql.Tx) error {
+				for _, event := range tc.Events {
+					err := dbHandler.DBWriteFailedEslEvent(ctx, transaction, &event)
+					if err != nil {
+						return err
+					}
+				}
+
+				actualEvents, err := dbHandler.DBReadLastFailedEslEvents(ctx, transaction, tc.Limit)
+				if err != nil {
+					return err
+				}
+
+				if len(actualEvents) > tc.Limit {
+					t.Fatalf("expected %d events, got %d", tc.Limit, len(actualEvents))
+				}
+
+				for i, actualEvent := range actualEvents {
+					if diff := cmp.Diff(tc.Events[len(tc.Events)-1-i].EslId, actualEvent.EslId); diff != "" {
```

Comment:
why do we need this logic to reverse the array index?
We should just expect the events to be in the correct order.
```suggestion
					if diff := cmp.Diff(tc.Events[i].EslId, actualEvent.EslId); diff != "" {
```